### PR TITLE
Typespecs

### PR DESCRIPTION
**Pull request by vanstee**
Based on #2 

Addes typespecs for all important public functions.

### CODE REVIEW

File: lib/fumanchu/compiler.ex [hunk 1]
```diff
@@ -3,11 +3,49 @@ defmodule FuManchu.Compiler do
   alias FuManchu.Parser
   alias FuManchu.Generator
 
+  @spec compile(String.t) :: (map -> String.t)
```
Comment (kevsmith) [inline lib/fumanchu/compiler.ex]: Defining a type for a compiled template would make this spec a bit easier to understand, imo.

``` elixir
@type compiled_template :: (%{} -> String.t)

@spec compile(String.t) :: compiled_template()
```

**Comment by kevsmith** [comment]
:+1:  modulo question re: typespec

File: lib/fumanchu/compiler.ex (hunk 1 above)
Comment (kevsmith) [inline lib/fumanchu/compiler.ex]: Hmm. Not entirely sure the spec, in either form, is accurate. Can compile return an error tuple or raise an error?

Returning error tuple:

``` elixir
@spec compile(String.t) :: compiled_template() | {:error, term()}
```

Raising an error:

``` elixir
@spec compile!(String.t) :: compiled_template() | no_return()
```

Comment (vanstee) [inline lib/fumanchu/compiler.ex]: It raises an error. To specify that do I just tack on an `| no_return`?

Comment (kevsmith) [inline lib/fumanchu/compiler.ex]: See my updated comment. If it raises an error probably should rename to `compile!`, yeah?

Comment (vanstee) [inline lib/fumanchu/compiler.ex]: Yeah, just made that change locally along with `render -> render!` :+1:

**Comment by kevsmith** [comment]
:+1:

**Pull request merged**